### DISCUSSION TITLE
Install eslint like a tool instead of like a library of the project

### DISCUSSION THREAD

**Discussion by rvitaliy** [eslint/eslint#15249] [Ideas]
Hi,
It's possible to install eslint like an external tool independent from the project dependency, since this is a tool and not a lib that we need to install in our project and use it's functionality, i mean my code works without eslint i need it only to check my code.

something like:
https://github.com/phpstan/phpstan
you can install it like a tool and run the command to analyse your code, without install it as dependency of your project.

**Comment by nzakas** [comment]
You can install ESLint globally if you’d like:

npm install eslint -g

Then you can run it as “eslint” in any directory.

**Reply by rvitaliy** [reply]
This is a valid option only if all your environments run in docker container, foreach project you have an isolated container and for this container you install eslint globally + all eslint plugins related for that project.
But this can't be an option if you don't use this approach.
each project have a different version of eslint & plugins, so globals is not an option in that case.

A different approach can be add tools directory in my project something like:
```
src/**myCode**
tools/eslint/package.json > with all configuration of eslint + plugins for this project
package.json > with "lint" script configuration something like "npx ./tools/eslint/node_modules/bin/eslint ./src/"
```
I tried this approach, but it's doesn't worked, I found some documentation that invite to install eslint and eslints plugins in project dependencies to allow it works.

**Reply by nzakas** [reply]
Sorry, ESLint is really designed to be bundled with projects at this point. Global installation is the only other option we have.
